src/test/java/org/jenkinsci/plugins/workflow/graph/FlowNodeTest.java

```diff
@@ -108,17 +117,338 @@
             }
         });
     }
-    private static void assertActiveSteps(WorkflowRun b, String... expected) {
+
+    private static void assertActiveSteps(WorkflowRun b, String... expected) throws Exception{
+        List<String> expectedList = Arrays.asList(expected);
+        Collections.sort(expectedList);
         List<String> actual = new ArrayList<>();
-        for (FlowNode n : new FlowGraphWalker(b.getExecution())) {
+        DepthFirstScanner scan = new DepthFirstScanner();
+        for (FlowNode n : scan.allNodes(b.getExecution())) {
             if (n.isActive()) {
                 String args = ArgumentsAction.getStepArgumentsAsString(n);
                 String name = n.getDisplayFunctionName();
                 actual.add(args != null ? name + ": " + args : name);
             }
         }
-        Collections.reverse(actual); // more readable
-        assertEquals(Arrays.asList(expected), actual);
+        Collections.sort(actual);
+        assertEquals(expectedList, actual);
+
+        // Now we clear the cache and try this again
+        Method m = FlowExecution.class.getDeclaredMethod("getInternalGraphLookup", null);
+        m.setAccessible(true);
+        Object ob = m.invoke((FlowExecution)(b.getExecution()), null);
```

Comment:
If `@Restricted(NoExternalUse.class) public`, could be called directly.

Reply:
Yeah, I'd considered that option